### ISSUE TITLE
Change bug report workflow.

### ISSUE THREAD

**Issue by neutrinog** [unfoldingWord-dev/ts-android#953]
Right now if the bug report fails to send due to a lack of internet or some other problem the dialog closes without a word to the user. We need to display an error to the user and ask them to try again. If the internet is not connect we need to ask them to turn it back on.

**Comment by RRohrssen** [comment]
Is there a document that describes the interaction(s) that should occur as we are sending or receiving data on their data connection?

**Comment by neutrinog** [comment]
no the only rules we are currently following is the user needs to confirm before doing anything that requires internet. And it should be clear that the action will utilize their internet.

**Comment by PhotoNomad0** [comment]
@neutrinog @cckozie - 
It looks like there are a couple of items here:
- prompt user before connecting to the internet and sending crash report.
- show message to user if sending crash report failed ask them to retry.

**Comment by cckozie** [comment]
I get the same Upload failed, Upload of feedback failed both when I have an Internet connection as well as when I don't.

**Comment by PhotoNomad0** [comment]
@cckozie - I wasn't clear enough in my question.  This looks like a workflow task.   What I gleaned from it was we need to implement:

- prompt user before connecting to the internet and sending crash report.
- show message to user if sending crash report failed ask them to retry.

Does this seem right to you?
@neutrinog

**Comment by cckozie** [comment]
@PhotoNomad0 I'd suggest changing the the Internet related verbiage on the Feedback dialog to the same as in desktop; "This will use your internet connection".
If it fails, it would be nice to tell the user if it was the Internet connection or something else (if we can tell), and then give them the retry option.

**Comment by cckozie** [comment]
There is now a retry option as well as a message indicating that there is not internet connection. I cannot fully test this until issue 1608 is fixed because sending feedback always fails in v11-153.

**Comment by cckozie** [comment]
I had tS in the background and was working in ES FIle Explorer when tS reported that it had crashed. I then turned off the wifi on the device and sent the crash report. It indicated that the feedback had been sent successfully, but with no Internet, obviously it was not. Just to make sure that it hadn't been cashed on the device, after I enabled the wifi I restarted tS and sent a feedback which did go through.
